src/Session/Session.cc

```diff
@@ -100,17 +101,13 @@ bool Session::_controller_deployment = false;
 std::thread* Session::_animation_thread = nullptr;
 
 Session::Session(uWS::WebSocket<false, true, PerSocketData>* ws, uWS::Loop* loop, uint32_t id, std::string address,
-    std::string top_level_folder, std::string starting_folder, std::shared_ptr<FileListHandler> file_list_handler, bool read_only_mode,
-    bool enable_scripting)
+    std::shared_ptr<FileListHandler> file_list_handler)
     : _socket(ws),
       _loop(loop),
       _id(id),
       _address(address),
-      _top_level_folder(top_level_folder),
-      _starting_folder(starting_folder),
-      _table_controller(std::make_unique<TableController>(_top_level_folder, _starting_folder)),
-      _read_only_mode(read_only_mode),
-      _enable_scripting(enable_scripting),
+      _table_controller(std::make_unique<TableController>(
+          ProgramSettings::GetInstance().top_level_folder, ProgramSettings::GetInstance().starting_folder)),
```

Comment:
These properties could be read from the global settings object inside the `TableController` constructor.